File: aggregator/entity/news.go
package entity

import "time"

type NewsHead struct {
	ID          string    `json:"id" bson:"_id,omitempty"`
	Title       string    `json:"title" bson:"title"`
	Description string    `json:"description" bson:"description"`
	Source      string    `json:"source" bson:"source"`
	PublishedAt time.Time `json:"published_at" bson:"published_at"`
}

type News struct {
	NewsHead   `bson:"inline"`
	Link       string   `json:"link" bson:"link"`
	Authors    []string `json:"authors" bson:"authors"`
	Tags       []string `json:"tags" bson:"tags"`
	Categories []string `json:"categories" bson:"categories"`
	Content    string   `json:"content" bson:"content"`
}

File: client/src/entities/news.ts
import type { NewsDTO, NewsHeadDTO } from "./news_dto"

export class NewsHead {
  id: string
  title: string
  description: string
  source: string
  publishedAt: Date

  constructor() {
    this.id = ""
    this.title = ""
    this.description = ""
    this.source = ""
    this.publishedAt = new Date()
  }

  static from(dto: NewsHeadDTO) {
    const obj = new NewsHead()
    obj.id = dto.id
    obj.title = dto.title
    obj.description = dto.description
    obj.source = dto.source
    obj.publishedAt = new Date(dto.published_at)

    return obj
  }
}

export class News extends NewsHead {
  link: string
  authors: string[]
  tags: string[]
  categories: string[]
  content: string

  constructor() {
    super()
    this.link = ""
    this.authors = []
    this.tags = []
    this.categories = []
    this.content = ""
  }

  static from(dto: NewsDTO) {
    const obj = new News()
    Object.assign(obj, { ...NewsHead.from(dto) })

    obj.link = dto.link
    obj.authors = dto.authors
    obj.tags = dto.tags
    obj.categories = dto.categories
    obj.content = dto.content

    return obj
  }
}
